src/router_core/delivery.c

```diff
@@ -1246,6 +1247,8 @@ qd_delivery_state_t *qdr_delivery_take_local_delivery_state(qdr_delivery_t *dlv,
 //
 void qdr_delivery_move_delivery_state_CT(qdr_delivery_t *dlv, qdr_delivery_t *peer)
 {
+    /// kag: crap, this may be an issue: possible drop of seq updates!
+
```

Comment:
For TCP_SEQ_ACK (or indeed PN_RECEIVED) it is just a sequence number (or two) to be updated. Perhaps we could update that in place?